lib/scope.go

```diff
@@ -174,6 +179,11 @@ func (s *scope) LocalResolver() (dsref.Resolver, error) {
 	return s.inst.resolverForMode("local")
 }
 
+// SetLogbook allows you to replace the current logbook with the given logbook
+func (s *scope) SetLogbook(l *logbook.Book) {
+	s.inst.logbook = l
+}
+
```

Comment:
In the `ProveProfileKey` path, we have one scenario where we actually replace the given logbook (now line 153 in `lib/registry.go`). This feels... weird, so would like some eyes on it.

Reply:
Yes, that is correct. When running `prove` the user is trying to reestablish an existing account (registered on cloud) to the current machine they're using. This involves pulling data down from cloud, including their old profileID, and replacing it in their current repo. They also need to get their logbook data from cloud and replace what was there before, so that their profileID agrees with their logbook which lets `push` and `pull` continue to work. Not doing so causes the "split logbook identity" problem.

If you read the definition of the `ProveProfileKey` function, you'll see that the `prove` command will fail if the repository contains more than zero refs. This means it can only run on a repository that has run `setup` but not yet run `save`, which will have a logbook that is basically empty (it will have the user op for the newly created profile, but that's all). In this case it's safe to replace the old logbook data, since it didn't have any important data in it anyway. In the future, it would be better to merge this temporary logbook data, treating it as a separate profile, but getting to that point probably requires finishing the logbook redesign first.

Reply:
great thank you for the detailed response!